### PR TITLE
Add keycloak instructions

### PR DESCRIPTION
**Pull request by ArendPeter**
This documents all the keycloak setup

It also updates the hardcoded frontend url

⚠️WARNING⚠️: This change will need some extra steps to be deployed

- [x]  Update keycloak ssl certification (at the moment it's self signed and axios will reject it)
- [ ]  Update heroku environment variables (heroku needs to point to the new keycloak environment variables, otherwise it'll be out of sync with the frontend keycloak url)

### CODE REVIEW

File: docs/keycloak_deployment.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+1. Go to the EC2 Console
+1. Select "Launch Instance"
+1. Configure as follows
+    * Name: Keycloak Server
+    * OS: Amazon Linux 2 64 Bit (I used ami-0cff7528ff583bf9a but you can use the latest)
+    * Instance Type: t2.micro (Keycloak needs at least .5 GB of memory and this is the cheapest one with 1 GB of memory)
+    * Key Pair: Create Key pair (RSA/pem), and download the pem file (keep the pem file in a safe you'll need it later to access the instance)
+    * Network Settings: Select "Allow HTTPs traffic from the internet"
+    * Configure Storage: Select advanced, 16 GB (default is 8 GB, that might be enough but it just seemed low to me), and set Encrypted to Yes, and KMS Key to aws/ebs
```
Comment (scottPlusPlus) [inline docs/keycloak_deployment.md:25]: do we know what we need to store?  Theoretically just the keycloak code + everyone's emails and passwords?

File: docs/keycloak_deployment.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+By default, EC2 instance will receive a new IP address on every reboot
+
+To keep it the same we need to allocate an elastic IP
+
+1. Within the EC2 console, go to the "Network & Security" section and select "Elastic IPs"
+1. Click "Allocate Elastic IP"
+1. Use the default options, and hit allocate
+1. Under Actions, select "Associate Elastic IP Address"
+1. In the instance section, select the Keycloak instance
+1. Click "Associate"
```
Comment (scottPlusPlus) [inline docs/keycloak_deployment.md:40]: maybe note that this ip will be used in several places in the keycloack configuration guide?

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -30,9 +30,9 @@ const App = () => {
   // http://keycloak.6j0.org/auth/realms/STAR%20Voting/.well-known/openid-configuration
   let keycloakBaseUrl;
   if(prodEndpoints.includes(window.location.origin)){
-    keycloakBaseUrl = 'https://keycloak.6j0.org/auth/realms/STAR%20Voting/protocol/openid-connect';
+    keycloakBaseUrl = 'https://52.205.245.149:8443/realms/STAR%20Voting/protocol/openid-connect';
   }else{
-    keycloakBaseUrl = 'https://keycloak.6j0.org/auth/realms/STAR%20Voting%20Dev/protocol/openid-connect';
+    keycloakBaseUrl = 'https://52.205.245.149:8443/realms/STAR%20Voting%20Dev/protocol/openid-connect';
```
Comment (scottPlusPlus) [inline frontend/src/App.tsx:35]: is this step included in the guide?

**Review by scottPlusPlus — APPROVED** [review]

File: docs/keycloak_deployment.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+nav_order: 3000
+---
+
+# Keycloak Deployment Guide
+
+Keycloak is the service we use for managing users and login sessions. Here we'll cover how to setup a fresh keycloak service from scratch
+
+## Get access to the STAR AWS account
+
+Contact Arend Peter ``@Arend Peter`` on slack to get a AWS user on the STAR AWS account
```
[the remark this answers is not in the document]
Comment (ArendPeter) [reply docs/keycloak_deployment.md:13]: Good point, I've setup admin accounts for you and Mike already, so I'll include both of you in the options as well. I'll also add some instructions for setting up new users

File: docs/keycloak_deployment.md (hunk 1 above)
Comment (ArendPeter) [reply docs/keycloak_deployment.md:25]: Yep, that's all for the moment, although I want to look into linking keycloak to a separate sql database

File: frontend/src/App.tsx (hunk 3 above)
Comment (ArendPeter) [reply frontend/src/App.tsx:35]: Yep, I mention "frontend/src/App.tsx" toward the end of the keycloak_configuration doc

**Review by mikefranze — CHANGES_REQUESTED** [review]
Needs conflicts resolved

**Comment by ArendPeter** [comment]
I updated the PR to address the comments and handle the git conflicts

**Review by mikefranze — APPROVED** [review]

**Comment by mikefranze** [comment]
Is this good to merge and push to heroku or do we still need to resolve ssl issue?

**Comment by ArendPeter** [comment]
> Is this good to merge and push to heroku or do we still need to resolve ssl issue?

Still need to add SSL, but I have the next steps now

I'll leave a comment when it's done (hopefully in the next few days)

**Comment by ArendPeter** [comment]
HTTPS is finally working!!! Thanks everyone for all the help along the way

This should be good to merge now

**Pull request merged**